lib/puppet/provider/mysql_user/mysql.rb

```diff
@@ -96,7 +96,9 @@ def create
 
   def destroy
     merged_name = @resource[:name].sub('@', "'@'")
-    self.class.mysql_caller("DROP USER '#{merged_name}'", 'system')
+    if_exists = (Puppet::Util::Package.versioncmp(mysqld_version, '5.7') >= 0) ? 'IF EXISTS ' : ''
```

Comment:
This needs a mariadb version check as well. Might as well throw in percona, too. e.g. https://github.com/xelmedia/puppetlabs-mysql/blob/b88fe42bb0302c4e4cd048a2b82d55c2c7e5f525/lib/puppet/provider/mysql_user/mysql.rb#L86-L87